lib/puppetserver/ca/action/generate.rb

```diff
@@ -296,7 +296,9 @@ def check_for_existing_ssl_files(certname, settings)
         end
 
         def process_alt_names(alt_names, certname)
-          return '' if alt_names.empty?
+          # It is recommended (and sometimes enforced) to always include
+          # the certname as a SAN, see RFC 2818 https://tools.ietf.org/html/rfc2818#section-3.1.
+          return "DNS:#{certname}" if alt_names.empty?
```

Comment:
What if there are other alt names? Wouldn't we still want to add the certname?

Reply:
We do. See below.

Reply:
Every other case I could find, we were already doing it.

Reply:
ahh, gotcha, didn't see the rest of this method 😅 
Do we have tests for those cases? I didn't see one near the test you added, so might be good to add.

Reply:
We test most of these already from several angles. I added one for specifically the line of code below this.